src/cudamatrix/cu-kernels-ansi.h

```diff
@@ -789,6 +789,9 @@ void cuda_uncompress_uint8(dim3 Gr, dim3 Bl, BaseFloat *dest,
                           MatrixDim dim, const uint8_t *src,
                           int src_stride, float scale);
 
+// Launches a kernel that does nothing, explicitly using the legacy default stream;
+// this will synchronize all threads without blocking.
+void cuda_legacy_noop();
```

Comment:
I'm sure you're aware already, but this is unused. 

I would consider renaming it to something like "cuda_synchronize_all_devices".

Consider adding a `cudaGetLastError()` to the end of the function definition. Up to you.

It also looks like you could do a for loop with `CuDevice::Instantiate::SelectGpuId(id)` and [cudaDeviceSynchronize](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__DEVICE.html#group__CUDART__DEVICE_1g10e20b05a95f638a4071a655503df25d).